### PR TITLE
PROJQUAY-1380 OIDC external authentication to use PREFERRED_URL_SCHEME in config.yaml

### PR DESCRIPTION
**Pull request by lazzarello**
**Issue:** https://issues.redhat.com/browse/PROJQUAY-1380

**Changelog:** 

Added: OIDC login uses value of PREFERRED_URL_SCHEME when redirecting after successful authentication.

**Testing:** 

Requires a custom external authentication OIDC provider to be configured and an installation to be behind a load balancer that is doing SSL termination.

**Details:** 

Flask's url_for() function ignores the value of PREFERRED_URL_SCHEME which will break OAuth2 authentication because it requires redirecting to only HTTPS endpoints.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,3 @@
-### master
-
```
Comment (thomasmckay) [inline CHANGELOG.md:2]: FYI - this should not be removed

Comment (lazzarello) [reply CHANGELOG.md:2]: lol, that's from starting at the qui-gon branch. :shrug: I'll put it back I guess.

**Comment by lazzarello** [comment]
@thomasmckay all CI passed. lmk if you need anything else to merge this.

**Comment by lazzarello** [comment]
I fixed another bug in this branch that references [PROJQUAY-1383](https://issues.redhat.com/browse/PROJQUAY-1383). An issue regarding a missing urlencoding function call.

**Comment by thomasmckay** [comment]
@jonathankingfc please review

**Review by jonathankingfc — APPROVED** [review]
LGTM

**Pull request merged**